packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3733,13 +3728,14 @@ export class ContainerRuntime
 		const result = await this.summarizerNode.refreshLatestSummary(
 			proposalHandle,
 			summaryRefSeq,
-			fetchLatestSnapshot,
-			readAndParseBlob,
-			summaryLogger,
 		);
 
+		if (result.latestSummaryUpdated && !result.wasSummaryTracked) {
```

Comment:
Can you please add a comment here explaning why in this case fetch snapshot is called.

Also nit: The `fetchLatestSnapshot()` const is no longer required. You can directly call `fetchSnapshotFromStorageAndClose` here.

Reply:
Yeah, I had left a comment about this too but VS Code ate it.  There's a ton of logic in there that doesn't necessarily make sense anymore, would be good to clean up here or in a follow-up.

Reply:
Most troubling is `waitForDeltaManagerToCatchup` which I'm pretty sure will hang if we're not already caught up (It should throw IMO since the Container is disposed). I wonder how often this is happening already.

Reply:
That wait function is also called in the `refreshLatestAck` codepath in `submitSummary` and this concern should be addressed there too

Reply:
Good point. It's a little weird that there is code running after the container runtime is explicitly disposed. For example, when `refreshLatestSummaryAckFromServer` is called from `submitSummary`, it would dispose the container runtime but would continue summarization and eventually fail when `checkContinue` is called later (and the failure reason would be different I believe). We should short circuit this whole thing and return a failure.

Reply:
Maybe that be cleaned up as part of AB#5417

Reply:
Cleaning up / short-circuiting later sounds fine, as long as we're sure we don't have a potential deadlock - I think we do.  That needs to be sorted out immediately.

Reply:
Added to [AB#5417](https://dev.azure.com/fluidframework/internal/_workitems/edit/5417)

> Good point. It's a little weird that there is code running after the container runtime is explicitly disposed. For example, when `refreshLatestSummaryAckFromServer` is called from `submitSummary`, it would dispose the container runtime but would continue summarization and eventually fail when `checkContinue` is called later (and the failure reason would be different I believe). We should short circuit this whole thing and return a failure.